resolver/bootstrap.go

```diff
@@ -48,7 +52,9 @@ func NewBootstrap(cfg *config.Config) (b *Bootstrap, err error) {
 		connectIPVersion: cfg.ConnectIPVersion,
 
 		systemResolver: net.DefaultResolver,
-		dialer:         &net.Dialer{},
+		dialer: &net.Dialer{
+			Timeout: defaultTimeout,
```

Comment:
I unfortunately can't make a suggestion above this block since the lines are unchanged. Can you add the following above the comment?
```go
		timeout := defaultTimeout
		if cfg.Upstreams.Timeout.IsAboveZero() {
			timeout = cfg.Upstreams.Timeout.ToDuration()
		}
```

And then we can use that here:
```suggestion
			Timeout: timeout,
```

Reply:
Changed as suggested